ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationSidebar.tsx

```diff
@@ -36,103 +54,62 @@ const platformConfigurationPaths = [
 ];
 
 type NavigationSidebarProps = {
-    hasReadAccess: HasReadAccess;
-    // eslint-disable-next-line react/no-unused-prop-types
-    isFeatureFlagEnabled: IsFeatureFlagEnabled;
+    isRenderedRoutePath: IsRenderedRoutePath;
 };
 
-function NavigationSidebar({ hasReadAccess }: NavigationSidebarProps): ReactElement {
-    const location: Location = useLocation();
+function NavigationSidebar({ isRenderedRoutePath }: NavigationSidebarProps): ReactElement {
+    const { pathname } = useLocation();
 
-    const vulnerabilityManagementPaths = [vulnManagementPath];
-    if (
-        hasReadAccess('VulnerabilityManagementRequests') ||
-        hasReadAccess('VulnerabilityManagementApprovals')
-    ) {
-        vulnerabilityManagementPaths.push(vulnManagementRiskAcceptancePath);
+    function isActiveFilter(routePath: string): boolean {
+        return routePath === vulnManagementPath
+            ? pathname === routePath
+            : (pathname as string).startsWith(routePath);
     }
-    if (hasReadAccess('VulnerabilityReports')) {
-        vulnerabilityManagementPaths.push(vulnManagementReportsPath);
+
+    const filteredPathsUnexpandable1 = unfilteredPathsUnexpandable1.filter(isRenderedRoutePath);
+    const filteredPathsVulnerabilityManagement =
+        unfilteredPathsVulnerabilityManagement.filter(isRenderedRoutePath);
+    const filteredPathsUnexpandable2 = unfilteredPathsUnexpandable2.filter(isRenderedRoutePath);
```

Comment:
Appending the numbers is a little obtuse.

Reply:
Replaced obtuse names with inline arrays of unexpandable paths.